topics/wallet-standard.md

```diff
@@ -0,0 +1,248 @@
+
+`appMetadata`: Information about the wallet.
+- `name` (`text`): The user-friendly name of the wallet.
+- `icon` (`text`, optional): An optional URL poiting to an icon resource representing the wallet.
+
+`networks`: A list of networks on which the user has agreed the relying party can operate. This should be a subset of the `networks` from the original request.
+- `chainId` (`text`): The chain ID of the network as described in the [CAIP-2]() standard.
+- `name` (`text`, optional): An optional user-friendly name of the network.
+- `rpcUrl` (`text`, optional): An optional custom RPC URL associated with the network.
+
+`scopes`: A list of permission scopes that the user has agreed the relying party can be granted. This should be a subset of the `scopes` field from the original request. Possible values:
+- `"canister_call"`
+
+`principal` (`blob`): The principal associated with the identity the user has selected.
+
+`ledger` (optional): Ledger data.
+- `subaccount` (`blob`, optional)
+
+`challenge` (`blob`): A challenge the wallet had to sign.
+
+`signature` (`blob`): The signature of the challenge generated by the wallet.
+
+#### Use-Case
+
+1. The relying party sends a `permission` request to the wallet.
+2. The wallet receives the message, presents the details of the to-be-established connection to the user and asks the user to select the identity that will be paired in response.
+    - If the user approves the request, the wallet sends a succesful response back to the relying party.
+    - If the user rejects the request, the wallet sends a response with an error back to the relying party.
+
+#### Example
+
+```json
+// Request
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "method": "permission",
+    "params": {
+        "version": "1",
+        "senderId": "" /* TODO */,
+        "appMetadata": {
+            "name": "My DApp"
+        },
+        "networks": [{
+            "chainId": "icp:737ba355e855bd4b61279056603e0550",
+        }],
+        "scopes": ["canister_call"],
+        "challenge": "" /* TODO */
+    }
+}
+
+// Response
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "result": {
+        "version": "1",
+        "senderId": "" /* TODO */,
+        "appMetadata": {
+            "name": "My Wallet"
+        },
+        "networks": [{
+            "chainId": "icp:737ba355e855bd4b61279056603e0550"
+        }],
+        "scopes": ["canister_call"],
+        "principal": "" /* TODO */,
+        "challenge": "" /* TODO */,
+        "signature": "" /* TODO */
+    }
+}
+```
+
+### Canister Call
+
+Once the connection between the relying party and wallet is established, the relying party can request the wallet to execute canister calls.
+
+#### Types
+
+- `text`: A plain `string` value.
+- `blob`: A `string` value describing binary data encoded in base64.
+- `nat`: A `string` value of an unsigned 64-bit integer.
+- `HashTree`: A recursive type defined as a union of:
```

Comment:
Looking at the definition of the type here, I think it might really be more practical to just treat certificate as a `blob`.